File: udyam/custom_script/customer/customer.js
frappe.ui.form.on("Customer", {

	refresh:function(frm){
		if(frm.doc.name){
			frm.set_df_property("is_supplier","read_only",1);

			frm.add_custom_button(__("Make Subscription"), function() {
				frappe.model.open_mapped_doc({
					method: "udyam.custom_script.customer.customer.make_subscription",
					frm: cur_frm
				});
			});
		}
	}	
});

File: udyam/custom_script/customer/customer.py
from __future__ import unicode_literals
import frappe
import json
from frappe.utils import flt, cstr, cint, today, getdate
from frappe.model.mapper import get_mapped_doc



@frappe.whitelist()
def make_subscription(source_name,target_doc=None):
	doclist = get_mapped_doc("Customer", source_name, {
		"Customer": {
			"doctype": "Subscription",
			"field_map": {
				"name": "customer"
			}
		},
	}, target_doc)
	return doclist

